lib/astutils.cpp

```diff
@@ -248,6 +248,23 @@ bool isDifferentKnownValues(const Token * const tok1, const Token * const tok2)
     return tok1->hasKnownValue() && tok2->hasKnownValue() && tok1->values() != tok2->values();
 }
 
+static bool isZeroBoundCond(const Token * const cond)
+{
+    if(cond == nullptr)
+        return false;
+    // Assume unsigned
+    bool isZero = cond->astOperand2()->getValue(0);
+    if (cond->str() == "==" || cond->str() == ">=")
+        return isZero;
+    if (cond->str() == "<=")
```

Comment:
I am curious how "<=" works in different situations it seems to always return true.

So this is a zero bound condition?

        if(s.size() <= 10)

Reply:
so I guess this condition should not be zero bounded?

    if (10 <= s.size())

Reply:
No, but this case wont happen because `isOppositeCond` only checks `astOperand1` for `.size()` function.